app/src/common/shared/org/mozilla/vrbrowser/browser/engine/SessionStack.java

```diff
@@ -803,8 +815,29 @@ protected void setTrackingProtection(final boolean aEnabled) {
             SessionState state = mSessions.get(mCurrentSession.hashCode());
             if (state != null && state.mSettings.isTrackingProtectionEnabled() != aEnabled) {
                 state.mSettings.setTrackingProtectionEnabled(aEnabled);
-                recreateSession(state.mSettings);
+                recreateSession(state);
+            }
+        }
+    }
+
+    public void clearCache() {
+        if (mRuntime != null) {
+            // Per GeckoView Docs:
+            // Note: Any open session may re-accumulate previously cleared data.
+            // To ensure that no persistent data is left behind, you need to close all sessions prior to clearing data.
+            // https://mozilla.github.io/geckoview/javadoc/mozilla-central/org/mozilla/geckoview/StorageController.html#clearData-long-
+            for (Map.Entry<Integer, SessionState> entry : mSessions.entrySet()) {
+                SessionState state = entry.getValue();
+                if (state != null) {
+                    state.mSession.stop();
+                    state.mSession.close();
+                }
             }
+
+            mRuntime.getStorageController().clearData(StorageController.ClearFlags.ALL).then(aVoid -> {
```

Comment:
This does more than just clear the cache. I think `StorageController.ALL_CACHES` would be better? If we want to purge all data I think it should be called something other than clear cache?

Reply:
Moved to Privacy and updated to ALL_CACHES.